pkg/scheduler/admission/pod_request_constraints.go

```diff
@@ -0,0 +1,85 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	admission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+func init() {
+	admission.RegisterPlugin("PodRequestConstraints", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return newPodRequestConstraints(pluginConfig), nil
+	})
+}
+
+// type nodeSpecifierConstraintConfig struct {
+// 	ProhibitNodeName      bool `json:"allowNodeName"`
+// 	AllowNodeSpecifier bool `json:"allowNodeSpecifier"`
+// }
+
+type podRequestConstraints struct {
+	*admission.Handler
+	config *PodRequestConstraintsConfig
+}
+
+// var _ = oadmission.WantsProjectCache(&podRequestConstraints{})
+// var _ = oadmission.WantsOpenshiftClient(&podRequestConstraints{})
+// var _ = oadmission.Validator(&podRequestConstraints{})
+
+func newPodRequestConstraints(config *PodRequestConstraintsConfig) admission.Interface {
+	return &podRequestConstraints{
+		config:  config,
+		Handler: admission.NewHandler(admission.Create),
+	}
+}
+
+func readConfig(reader io.Reader) (*PodRequestConstraintsConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return &PodRequestConstraintsConfig{}, nil
+	}
+
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	config := &PodRequestConstraintsConfig{}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	// No validation needed since config is just 2 bools
+	return config, nil
+}
+
+func (oo *podRequestConstraints) Admit(aa admission.Attributes) error {
+	if oo.config == nil {
+		return nil
+	}
+	if aa.GetResource() != kapi.Resource("pods") {
+		return nil
+	}
+	if len(aa.GetSubresource()) > 0 {
```

Comment:
For the purpose of this PR, I don't believe that we want to intercept the subresource that the scheduler uses. We just want to prevent the case of a regular user that's creating a pod and trying to set where that pod should be scheduled. For that, subresource == "" I believe should be the right check.